lib/improver/ensemble_copula_coupling/ensemble_copula_coupling.py

```diff
@@ -353,23 +359,45 @@ def _add_bounds_to_thresholds_and_probabilities(
                     each end.
         """
         lower_bound, upper_bound = bounds_pairing
-        threshold_points = insert_lower_and_upper_endpoint_to_1d_array(
-            threshold_points, lower_bound, upper_bound)
+        threshold_points_with_endpoints = \
+            insert_lower_and_upper_endpoint_to_1d_array(
+                threshold_points, lower_bound, upper_bound)
         probabilities_for_cdf = concatenate_2d_array_with_2d_array_endpoints(
             probabilities_for_cdf, 0, 1)
-        if np.any(np.diff(threshold_points) < 0):
+
+        if np.any(np.diff(threshold_points_with_endpoints) < 0):
             msg = ("The end points added to the threshold values for "
                    "constructing the Cumulative Distribution Function (CDF) "
                    "must result in an ascending order. "
                    "In this case, the threshold points {} must be "
                    "outside the allowable range given by the "
-                   "bounds {}".format(
-                       threshold_points, bounds_pairing))
-            raise ValueError(msg)
-        return threshold_points, probabilities_for_cdf
+                   "bounds {}. ".format(threshold_points_with_endpoints,
+                                        bounds_pairing))
```

Comment:
I know this isn't your change but the error message here seems a bit convoluted / unclear.  Can we clarify the problem to which it is referring?  (Which I think is that the thresholds calculated from the data have exceeded the permitted CDF bounds, but we'd better check that to be sure.)

Reply:
I have updated this message. The trigger for the error is the threshold values not being in ascending order for the CDF, this as a result of the threshold values exceeding the bounds. I think it's important to mention both these points and useful to see the values that have caused the error.